### PR TITLE
Fixed Flaky Unit tests

### PR DESCRIPTION
**Pull request by shubhamdixit863**
No

### Modifications

Modified Some tests  in `util_test.go`  to fix flaky unit tests.

### Verification

By running unit tests `make test`

### CODE REVIEW

**Review by dpadhiar — APPROVED** [review]
LGTM - will wait for @xdevxy to merge

File: internal/git/util_test.go [hunk 1]
```diff
@@ -453,7 +453,7 @@ AAAECl1AymWUHNdRiOu2r2dg97arF3S32bE5zcPTqynwyw50HAtto0bVGTAUATJhiDTjKa
 	repo, err := cloneRepo(context.Background(), gitSync, cloneOptions, metric)
 	assert.NoError(t, err)
 	assert.NotNil(t, repo)
-	err = FileExists(repo, "data.yaml") // data.yaml default file exists in docker git
+	err = FileExists(repo, "readme.md") // readme.md default file exists in docker git
```
Comment (xdevxy) [inline internal/git/util_test.go:456]: This is working because we always have `readme.md` by default?

**Review by xdevxy — APPROVED** [review]

**Comment by xdevxy** [comment]
@shubhamdixit863 merged to unblock, however has one question https://github.com/numaproj-labs/numaplane/pull/275#discussion_r1589815084, Thanks!

File: internal/git/util_test.go (hunk 1 above)
Comment (shubhamdixit863) [reply internal/git/util_test.go:456]: yes ,as we are not putting any yaml files . For bdd tests they are being put by git pod which does got operations . So for testing purpose have added this file

**Pull request merged**